lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -518,6 +715,69 @@ def _call(self, method: Callable, json_data: Dict[AnyStr, Any]):
             )
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
+    def _check_authorization(self, user: str, access: str):
+        """
+        Check whether an API call is able to access data, based on the API's
+        authorization header, the requested user, the requested access
+        policy, and the API's role.
+
+        If "user" is None, then the request is unauthenticated. READ operations
+        may be allowed, UPDATE and DELETE operations will not be allowed.
+
+        If "access" is None, READ will assume we're looking for access only to
+        public datasets.
+
+        for API_OPERATION.READ:
+
+            Any call, with or without an authenticated user token, can access
+            public data.
+
+            Any authenticated user can access their own private data.
+
+            Any authenticated ADMIN user can access any private data.
+
+        for API_OPERATION.UPDATE:
+
+            An authenticated user is required.
+
+            Any authenticated user can update their own data.
+
+            Any authenticated ADMIN user can update any data.
+
+        Args:
+            user: The user parameter to the API, or None
+            access: The access parameter to the API, or None
+
+        Raises:
+            UnauthorizedAccess The user isn't authorized for the requested
+                access.
+        """
+        authorized_user: User = Auth.token_auth.current_user()
+        authorized = True
+        self.logger.debug(
+            "Authorizing {} access for {} to user {} with access {}",
+            self.role,
+            authorized_user,
+            user,
+            access,
+        )
+        if self.role != API_OPERATION.READ or access == Dataset.PRIVATE_ACCESS:
+            if authorized_user is None:
+                self.logger.warning(
+                    "Attempt to {} user {} data without login", self.role, user
+                )
+                authorized = False
+            elif user != authorized_user.username and not authorized_user.is_admin():
+                self.logger.warning(
+                    "Unauthorized attempt by {} to {} user {} data",
+                    authorized_user,
+                    self.role,
+                    user,
+                )
+                authorized = False
+        if not authorized:
+            raise UnauthorizedAccess(authorized_user, self.role, user, access)
+
```

Comment:
Do we have a unit test just for this method?

If so, does it cover all possible inputs?  Seems like this is the function we need to be rock solid.

Reply:
Wait, adding a unit test for this does not require a framework for unit tests, right?  It would be good to track this as a separate issue to be handled independent of our other framework needs.